edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/serviceaccount/KubernetesServiceAccountByValueEqualityComparer.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment.ServiceAccount
+{
+    using System.Collections.Generic;
+    using k8s.Models;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public sealed class KubernetesServiceAccountByValueEqualityComparer : IEqualityComparer<V1ServiceAccount>
+    {
+        static readonly DictionaryComparer<string, string> MetadataComparer = DictionaryComparer.StringDictionaryComparer;
+
+        public bool Equals(V1ServiceAccount x, V1ServiceAccount y)
+        {
+            if (ReferenceEquals(x, y))
+            {
+                return true;
+            }
+
+            if (ReferenceEquals(x, null))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(y, null))
+            {
+                return false;
+            }
+
+            if (x.GetType() != y.GetType())
+            {
+                return false;
+            }
+
+            return x.Metadata.Name == y.Metadata.Name &&
```

Comment:
Should we care about namespace here? I mean if a resource from another namespace accidently got here...

Reply:
It's not set in KubernetesServiceAccountMapper , so we don't always have it here.